dashboard/src/components/AppList/AppList.tsx

```diff
@@ -59,7 +59,11 @@ class AppList extends React.Component<IAppListProps, IAppListState> {
       namespace,
     } = this.props;
     // refetch if new namespace or error removed due to location change
-    if (prevProps.namespace !== namespace || (!error && prevProps.apps.error)) {
+    if (
+      prevProps.namespace !== namespace ||
+      prevProps.cluster !== cluster ||
+      (!error && prevProps.apps.error)
+    ) {
```

Comment:
there are a lot of comparisons like this one:

```
dashboard/src/components/OperatorNew/OperatorNew.tsx
63:    if (prevProps.namespace !== this.props.namespace) {

dashboard/src/components/OperatorList/OperatorList.tsx
88:    if (prevProps.namespace !== this.props.namespace) {

dashboard/src/components/OperatorView/OperatorView.tsx
40:    if (prevProps.namespace !== this.props.namespace) {

dashboard/src/components/AppView/AppView.tsx
93:    if (prevProps.namespace !== namespace) {

dashboard/src/components/AppList/AppList.tsx
62:    if (prevProps.namespace !== namespace || (!error && prevProps.apps.error)) {

dashboard/src/components/ServiceInstanceList/ServiceInstanceList.tsx
76:      (prevProps.namespace !== namespace || (error && !prevProps.error))

dashboard/src/components/Config/AppRepoList/AppRepoList.tsx
100:    if (prevProps.namespace !== namespace || (prevProps.errors.fetch && !fetch)) {

dashboard/src/components/OperatorInstance/OperatorInstance.tsx
62:    if (prevProps.namespace !== namespace) {

dashboard/src/components/ServiceInstanceView/ServiceInstanceView.tsx
94:    if (prevProps.namespace !== namespace) {

```

It is a pain to maintain those... using function components help so if you miss something I should be able to notice it in the new UI.

Reply:
I updated the two relevant ones in 12628de (appview and apprepolist), as we're not yet supporting operators across clusters.